app/selectors/autocomplete.js

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {General} from 'mattermost-redux/constants';
+import {getMyChannels, getOtherChannels} from 'mattermost-redux/selectors/entities/channels';
+import {
+    getCurrentUser, getProfilesInCurrentChannel,
+    getProfilesNotInCurrentChannel, getProfilesInCurrentTeam
+} from 'mattermost-redux/selectors/entities/users';
+import {sortChannelsByDisplayName} from 'mattermost-redux/utils/channel_utils';
+import {sortByUsername} from 'mattermost-redux/utils/user_utils';
+
+import {getCurrentUserLocale} from 'app/selectors/i18n';
+
+export const filterMembersInChannel = createSelector(
+    getProfilesInCurrentChannel,
+    (state, opts) => opts,
+    (profilesInChannel, opts) => {
+        const {array} = opts;
+        array.splice(0, array.length);
+
+        let profiles;
+        const matchTerm = opts.matchTerm;
+        if (matchTerm) {
+            profiles = profilesInChannel.filter((p) => {
+                return ((p.id !== opts.currentUserId) && (
```

Comment:
Why is this passed through opts? We could just use the selector for that directly